src/ERC20ForwarderClient.js

```diff
@@ -509,6 +665,28 @@ class ERC20ForwarderClient {
    */
   async sendTxPersonalSign({ req, signature = null, userAddress }) {
     try {
+      // should we?
+      // check allowance at each final call
+      // needs spendValue that is cost
+
+      /*if(!userAddress)
+        userAddress = await this.provider.getSigner().getAddress();
+
+      const allowedToSpend = await this.erc20ForwarderApproved(
```

Comment:
Why this was commented?

Reply:
it wasnt there in sendTx methods
cause, allowedToSpend is checked in buildTx


now there are two ways (the problem/reason is buildTx shouldnt check allowed to spend and reject if it is meant for permit chained execution) 

i) move allowance check to sendTx stages from buildTx
ii) in buildTx check based on permitType (whether its passed or not) then only check allowance  

ii is currently done now. Also increased cost is displayed if permitType is passed

Reply:
any thoughts? check buildTx changes also in a file